src/renderers/dom/shared/HTMLDOMPropertyConfig.js

```diff
@@ -209,6 +229,54 @@ var HTMLDOMPropertyConfig = {
   },
   DOMPropertyNames: {
   },
+  DOMMutationMethods: {
+    value: function(node, next) {
```

Comment:
Added, and I brought over the `<progress>` test case from #7359 too. Though it re-introduces the `input[type="number"]` decimal place loss issue. Either way both PRs are synced.

I don't feel awesome about this `value` mutation method, but with all of the mutations in one place, it's way easier to reason about  the `[type="number"]` issues on this branch. Curious if I'll get any inspiration shifting back here.